OmsAgent/omsagent.py

```diff
@@ -64,59 +84,252 @@ def main():
+                                                                  optionalParams)
+
+    hutil_log_info(hutil, 'Handler initiating onboarding.')
+    exit_code, output = run_get_output(onboard_cmd)
+    # To avoid exposing the shared key, print output separately
+    hutil_log_info(hutil, 'Output of onboarding command: \n{0}'.format(output))
+    return exit_code
+
+
+def disable(hutil):
+    """
+    Disable all OMS workspace processes on the machine.
+    """
+    # Check if the service control script is available
+    if not os.path.exists(OmsAgentServiceScript):
+        log_and_exit(hutil, 'Disable', 1, 'OMSAgent service control script ' \
+                                          '{0} does not exist. Disable ' \
+                                          'cannot be called before ' \
+                                          'install.'.format(
+                                                     OmsAgentServiceScript))
+        return 1
+
+    cmd = DisableOmsAgentServiceCommandTemplate.format(OmsAgentServiceScript)
+    exit_code, output = run_command_and_log(hutil, cmd)
+    return exit_code
+
+
+# Dictionary of operations strings to methods
+operations = {'Disable' : disable,
+              'Uninstall' : uninstall,
+              'Install' : install,
+              'Enable' : enable,
+              # Upgrade is noop since omsagent.py->install() will be called
+              # everytime upgrade is done due to upgradeMode =
+              # "UpgradeWithInstall" set in HandlerManifest
+              'Update' : dummy_command
+}
+
+
 def parse_context(operation):
-    hutil = Util.HandlerUtility(waagent.Log, waagent.Error)
-    hutil.do_parse_context(operation)
+    """
+    Initialize a HandlerUtil object for this operation.
+    If the required modules have not been imported, this will return None.
+    """
+    hutil = None
+    if 'Utils.WAAgentUtil' in sys.modules and 'Utils.HandlerUtil' in sys.modules:
+        try:
+            hutil = HUtil.HandlerUtility(waagent.Log, waagent.Error)
+            hutil.do_parse_context(operation)
+        # parse_context may throw KeyError if necessary JSON key is not
+        # present in settings
+        except KeyError as e:
+            waagent_log_error('Unable to parse context with error: ' \
+                              '{0}'.format(e.message))
+            raise OmsAgentParameterMissingError
     return hutil
 
 
 def is_vm_supported_for_extension():
-    '''
-    Returns for platform.linux_distribution() vary widely in format, such as '7.3.1611' returned
-    for a machine with CentOS 7, so the first provided digits must match
-    '''
+    """
+    Checks if the VM this extension is running on is supported by OMSAgent
+    Returns for platform.linux_distribution() vary widely in format, such as
+    '7.3.1611' returned for a machine with CentOS 7, so the first provided
+    digits must match
+    Though Ubuntu 16.10 is not officially supported, we will allow it to
```

Comment:
Please remove this sentence "Though.." it can be confusing, since this was on azure Linux VM extension supported list for at least a year.